pkg/middleware/auth_test.go

```diff
@@ -201,9 +206,143 @@ var _ = Describe("Auth", func() {
 				fakeFiatClient.AuthorizeReturns(fakeResp, nil)
 			})
 
-			It("returns status Forbidden", func() {
+			It("returns status OK", func() {
 				Expect(c.Writer.Status()).To(Equal(http.StatusOK))
 			})
 		})
 	})
+
+	Describe("#FilterAuthorizedApplications", func() {
+		BeforeEach(func() {
+			hf = PostFilterAuthorizedApplications("READ")
+			allApps = append(allApps, core.Application{
+				Name: "test-app1",
+			})
+			c.Set(core.KeyAllApplications, allApps)
+		})
+
+		JustBeforeEach(func() {
+			hf(c)
+		})
+
+		AfterEach(func() {
+			c.Abort()
+		})
+
+		When("There is an error attached to the context", func() {
+			BeforeEach(func() {
+				c.Errors = append(c.Errors, c.Error(errors.New("fake error")))
+			})
+
+			It("returns", func() {
+
+			})
+
+		})
+
+		When("user is missing from header", func() {
+			BeforeEach(func() {
+				r.Header.Del("X-Spinnaker-User")
+			})
+
+			It("returns the list of all apps without filtering", func() {
+				Expect(c.Writer.Status()).To(Equal(http.StatusOK))
+			})
+		})
+
+		When("fiate Authorize function returns an error", func() {
```

Comment:
There's a typo here, and I think it would be better as just `When("fiat returns an error", func() {`.